File: tests/workflow/producer/typescript/src/producer/implementation/events/events.sub.eventa.ts
import {
    Producer,
    Identification,
    Filter,
    broadcastAll,
    Context,
    Protocol,
} from "./index";
import { emit } from "../../events/events.sub.eventa";

export class Output {
    static REQUIRED = [    
        Protocol.GroupB.GroupC.StructA,
        Protocol.GroupB.GroupC.StructB,
    ];
    private _broadcasts: Array<[string[], Protocol.Convertor<any>]> = [];

    public broadcast(uuids: string[]): {        
        GroupBGroupCStructA(msg: Protocol.GroupB.GroupC.StructA): Output;
        GroupBGroupCStructB(msg: Protocol.GroupB.GroupC.StructB): Output;
    } {
        const self = this;
        return {            
            GroupBGroupCStructA(msg: Protocol.GroupB.GroupC.StructA): Output {
                if (
                    self._broadcasts.find(
                        (b) =>
                            b[1].getId() ===
                            Protocol.GroupB.GroupC.StructA.getId()
                    ) !== undefined
                ) {
                    throw new Error(
                        `Broadcast Protocol.GroupB.GroupC.StructA already has been defined.`
                    );
                }
                self._broadcasts.push([uuids, msg]);
                return self;
            },
            GroupBGroupCStructB(msg: Protocol.GroupB.GroupC.StructB): Output {
                if (
                    self._broadcasts.find(
                        (b) =>
                            b[1].getId() ===
                            Protocol.GroupB.GroupC.StructB.getId()
                    ) !== undefined
                ) {
                    throw new Error(
                        `Broadcast Protocol.GroupB.GroupC.StructB already has been defined.`
                    );
                }
                self._broadcasts.push([uuids, msg]);
                return self;
            },
        };
    }

	public error(): Error | undefined {
		let error: Error | undefined;
		Output.REQUIRED.forEach((ref) => {
			if (error !== undefined) {
				return;
			}
			if (
				this._broadcasts.find((msg) => {
					return msg[1].getId() === ref.getId();
				}) === undefined
			) {
				error = new Error(
					`Broadcast ${ref.getSignature()}/${ref.getId()} is required, but hasn't been found`
				);
			}
		});
		return error;
	}

    public broadcasts(): Array<[string[], Protocol.Convertor<any>]> {
        return this._broadcasts;
    }
}

export function handler(
    event: Protocol.Events.Sub.EventA,
    filter: Filter,
    context: Context,
    producer: Producer
): Promise<void> {
    return emit(event, filter, context, producer).then((output) => {
        const error: Error | undefined = output.error();
        if (error instanceof Error) {
            return Promise.reject(error);
        }
        return broadcastAll(producer, output.broadcasts());
    });
}

File: tests/workflow/producer/rust/src/producer/implementation/emitters/events_sub_eventa.rs
use super::{
    broadcast, events, identification, producer::Control, protocol, unbound_pack, Context,
    EmitterError, scope::AnonymousScope,
};
use clibri::server;
use uuid::Uuid;

pub async fn emit<E: server::Error, C: server::Control<E>>(
    event: protocol::Events::Sub::EventA,
    filter: &identification::Filter<'_>,
    context: &mut Context,
    control: &Control<E, C>,
) -> Result<(), EmitterError> {
    let mut scope: AnonymousScope<'_, E, C> = AnonymousScope::new(context, control, filter);
    let mut broadcasting: Vec<(Vec<Uuid>, Vec<u8>)> = vec![];
    let (mut broadcast_groupb_groupc_structa, mut broadcast_groupb_groupc_structb) =
        events::events_sub_eventa::emit(event, &mut scope)
            .await
            .map_err(EmitterError::Emitting)?;
    broadcasting.push((
        broadcast_groupb_groupc_structa.0,
        unbound_pack(&0, &mut broadcast_groupb_groupc_structa.1)?,
    ));
    broadcasting.push((
        broadcast_groupb_groupc_structb.0,
        unbound_pack(&0, &mut broadcast_groupb_groupc_structb.1)?,
    ));
    for msg in broadcasting.iter_mut() {
        broadcast::<E, C>(msg, control).await?;
    }
    scope.call().await;
    Ok(())
}